lib/public/utilities/fetch/DataSourceTransformer.js

```diff
@@ -0,0 +1,57 @@
+/**
+ *  @license
+ *  Copyright CERN and copyright holders of ALICE O2. This software is
+ *  distributed under the terms of the GNU General Public License v3 (GPL
+ *  Version 3), copied verbatim in the file "COPYING".
+ *
+ *  See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ *  In applying this license CERN does not waive the privileges and immunities
+ *  granted to it by virtue of its status as an Intergovernmental Organization
+ *  or submit itself to any jurisdiction.
+ */
+import { ObservableData } from '../ObservableData.js';
+import { DataSource } from '../DataSource.js';
+
+/**
+ * Class to abstract notifying from data transformation
+ * @template U, V
+ */
+export class DataSourceTransfomer extends DataSource {
```

Comment:
A transformer is almost never a subclass of what it transforms. For me a transformer should simply have a function named `transform` (actually the name is, I think, more commonly `adapter`)